src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1981,10 +1981,14 @@ protected function get_available_actions( $post, $request ) {
 
 		$post_type = get_post_type_object( $post->post_type );
 
-		if ( 'attachment' !== $this->post_type && current_user_can( $post_type->cap->publish_posts ) ) {
+		if ( 'attachment' !== $this->post_type && current_user_can( 'publish_post', $post->ID ) ) {
```

Comment:
Since this is an action for the specific post we should use the meta cap and pass the post ID to the check. By default this just [falls back to `publish_posts`](https://github.com/WordPress/wordpress-develop/blob/90d8fdb5bb9a58b3bf1d56749bf9634edd0dbd79/src/wp-includes/capabilities.php#L264-L279).

Reply:
Is this related to https://core.trac.wordpress.org/ticket/47443?